### DISCUSSION TITLE
FATAL: container creation failed with singularity profile

### DISCUSSION THREAD

**Discussion by nhhaidee** [nextflow-io/nextflow#3914] [Q&A]
## Bug report 

Can not run pipeline with singularity profile

### Expected behavior and actual behavior

Run pipeline (https://github.com/CFIA-NCFAD/nf-flu) with singularity profile but it failed with error 

### Steps to reproduce the problem

Run pipeline with `-profile singularity`

### Program output 

`Command error:
  FATAL:   container creation failed: mount /proc/self/fd/3->/usr/local/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/3: failed to find loop device: could not attach image file to loop device: no loop devices available
`

### Environment 

* Nextflow version: 23.04.1.5866
* Java version: openjdk 11.0.9.1-internal 2020-11-04
* Operating system: Ubuntu 22.04
* Bash version: 5.1.16(1)-release (x86_64-pc-linux-gnu)

### Additional context

There is no problem with Docker profile

**Comment by nhhaidee** [comment]
The loop devices seem to be OK

`dev/loop0    /dev/loop109  /dev/loop12   /dev/loop130  /dev/loop141  /dev/loop152  /dev/loop163  /dev/loop174  /dev/loop185  /dev/loop196  /dev/loop206  /dev/loop217  /dev/loop228  /dev/loop239  /dev/loop25   /dev/loop30  /dev/loop41  /dev/loop52  /dev/loop63  /dev/loop74  /dev/loop85  /dev/loop96
/dev/loop1    /dev/loop11   /dev/loop120  /dev/loop131  /dev/loop142  /dev/loop153  /dev/loop164  /dev/loop175  /dev/loop186  /dev/loop197  /dev/loop207  /dev/loop218  /dev/loop229  /dev/loop24   /dev/loop250  /dev/loop31  /dev/loop42  /dev/loop53  /dev/loop64  /dev/loop75  /dev/loop86  /dev/loop97
/dev/loop10   /dev/loop110  /dev/loop121  /dev/loop132  /dev/loop143  /dev/loop154  /dev/loop165  /dev/loop176  /dev/loop187  /dev/loop198  /dev/loop208  /dev/loop219  /dev/loop23   /dev/loop240  /dev/loop251  /dev/loop32  /dev/loop43  /dev/loop54  /dev/loop65  /dev/loop76  /dev/loop87  /dev/loop98
/dev/loop100  /dev/loop111  /dev/loop122  /dev/loop133  /dev/loop144  /dev/loop155  /dev/loop166  /dev/loop177  /dev/loop188  /dev/loop199  /dev/loop209  /dev/loop22   /dev/loop230  /dev/loop241  /dev/loop252  /dev/loop33  /dev/loop44  /dev/loop55  /dev/loop66  /dev/loop77  /dev/loop88  /dev/loop99
/dev/loop101  /dev/loop112  /dev/loop123  /dev/loop134  /dev/loop145  /dev/loop156  /dev/loop167  /dev/loop178  /dev/loop189  /dev/loop2    /dev/loop21   /dev/loop220  /dev/loop231  /dev/loop242  /dev/loop253  /dev/loop34  /dev/loop45  /dev/loop56  /dev/loop67  /dev/loop78  /dev/loop89  /dev/loop-control
/dev/loop102  /dev/loop113  /dev/loop124  /dev/loop135  /dev/loop146  /dev/loop157  /dev/loop168  /dev/loop179  /dev/loop19   /dev/loop20   /dev/loop210  /dev/loop221  /dev/loop232  /dev/loop243  /dev/loop254  /dev/loop35  /dev/loop46  /dev/loop57  /dev/loop68  /dev/loop79  /dev/loop9
/dev/loop103  /dev/loop114  /dev/loop125  /dev/loop136  /dev/loop147  /dev/loop158  /dev/loop169  /dev/loop18   /dev/loop190  /dev/loop200  /dev/loop211  /dev/loop222  /dev/loop233  /dev/loop244  /dev/loop255  /dev/loop36  /dev/loop47  /dev/loop58  /dev/loop69  /dev/loop8   /dev/loop90
/dev/loop104  /dev/loop115  /dev/loop126  /dev/loop137  /dev/loop148  /dev/loop159  /dev/loop17   /dev/loop180  /dev/loop191  /dev/loop201  /dev/loop212  /dev/loop223  /dev/loop234  /dev/loop245  /dev/loop26   /dev/loop37  /dev/loop48  /dev/loop59  /dev/loop7   /dev/loop80  /dev/loop91
/dev/loop105  /dev/loop116  /dev/loop127  /dev/loop138  /dev/loop149  /dev/loop16   /dev/loop170  /dev/loop181  /dev/loop192  /dev/loop202  /dev/loop213  /dev/loop224  /dev/loop235  /dev/loop246  /dev/loop27   /dev/loop38  /dev/loop49  /dev/loop6   /dev/loop70  /dev/loop81  /dev/loop92
/dev/loop106  /dev/loop117  /dev/loop128  /dev/loop139  /dev/loop15   /dev/loop160  /dev/loop171  /dev/loop182  /dev/loop193  /dev/loop203  /dev/loop214  /dev/loop225  /dev/loop236  /dev/loop247  /dev/loop28   /dev/loop39  /dev/loop5   /dev/loop60  /dev/loop71  /dev/loop82  /dev/loop93
/dev/loop107  /dev/loop118  /dev/loop129  /dev/loop14   /dev/loop150  /dev/loop161  /dev/loop172  /dev/loop183  /dev/loop194  /dev/loop204  /dev/loop215  /dev/loop226  /dev/loop237  /dev/loop248  /dev/loop29   /dev/loop4   /dev/loop50  /dev/loop61  /dev/loop72  /dev/loop83  /dev/loop94
/dev/loop108  /dev/loop119  /dev/loop13   /dev/loop140  /dev/loop151  /dev/loop162  /dev/loop173  /dev/loop184  /dev/loop195  /dev/loop205  /dev/loop216  /dev/loop227  /dev/loop238  /dev/loop249  /dev/loop3    /dev/loop40  /dev/loop51  /dev/loop62  /dev/loop73  /dev/loop84  /dev/loop95
`

**Comment by Biocentric** [comment]
We have the same issue, singularity profile does not work on our cluster system. We looked at the solution proposed here:
https://github.com/sylabs/singularity/issues/1499

But it does not yield a consistent solution for Nextflow. Test singularity containers will run after this intervention but no full Nextflow runs.

**Comment by Fiwx** [comment]
I have the same issue. Only test runs will work, but not full Nextflow runs. Here is the error:

  ```FATAL:   container creation failed: mount /proc/self/fd/4->/usr/local/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/4: failed to find loop device: could not attach image file to loop device: no loop devices available```
